pkg/image/controller/controller_test.go

```diff
@@ -606,6 +606,15 @@ func TestControllerWithSpecTags(t *testing.T) {
 						},
 					},
 				},
+				{
```

Comment:
You need test cases verifying that retries are being returned by those methods when you expect.  I'd like to see much more on that before we merge this.